lib/archethic/shared_secrets/node_renewal.ex

```diff
@@ -50,10 +50,19 @@ defmodule Archethic.SharedSecrets.NodeRenewal do
   @spec next_authorized_node_public_keys() :: list(Crypto.key())
   def next_authorized_node_public_keys do
     DB.get_latest_tps()
-    |> Election.next_authorized_nodes(P2P.available_nodes())
+    |> Election.next_authorized_nodes(candidates(), P2P.authorized_nodes())
     |> Enum.map(& &1.last_public_key)
```

Comment:
It should be `first_public_key` because if node disconnect and reconnect, it will have a new last public key but during disconnection time, other nodes don't know the new public key of the node.
And so when the node reconnect it will not be able to descrypt the secret as it's last public key is not the same than the one added by others nodes in the transaction.
By updating this, node shared secrets transaction have to be decrypted with node first public key